driver-core/src/main/com/mongodb/connection/netty/ReadTimeoutHandler.java

```diff
@@ -40,13 +40,16 @@ public ReadTimeoutHandler(final long readTimeout) {
     }
 
     void scheduleTimeout(final ChannelHandlerContext ctx) {
-        timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        if (timeout == null) {
+            timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        }
     }
 
     void removeTimeout(final ChannelHandlerContext ctx) {
         if (ctx.channel().eventLoop().inEventLoop()) {
             if (timeout != null) {
```

Comment:
we will always will be in the event loop for `removeTimeout`.  As we schedule it via the event loop executor in the `NettyStream`.  

Thats is there to ensure that if accidentally called outside the loop it wont break - we can change that to an assertion and it will break but not where anyone expects.